icu4c/source/common/lstmbe.cpp

```diff
@@ -403,24 +402,21 @@ LSTMData::LSTMData(UResourceBundle* rb, UErrorCode &status)
         fType = GRAPHEME_CLUSTER;
     }
     fName = ures_getStringByKey(rb, "model", nullptr, &status);
-    fDataRes = ures_getByKey(rb, "data", nullptr, &status);
+    LocalUResourceBundlePointer dataRes(ures_getByKey(rb, "data", nullptr, &status));
+    LocalUResourceBundlePointer dictRes(ures_getByKey(rb, "dict", nullptr, &status));
     if (U_FAILURE(status)) return;
     int32_t data_len = 0;
-    const int32_t* data = ures_getIntVector(fDataRes, &data_len, &status);
-    if (U_FAILURE(status)) return;
-    fDictRes = ures_getByKey(rb, "dict", nullptr, &status);
-    if (U_FAILURE(status)) return;
-    U_ASSERT(fDictRes != nullptr);
-    int32_t num_index = ures_getSize(fDictRes);
+    const int32_t* data = ures_getIntVector(dataRes.getAlias(), &data_len, &status);
+    int32_t num_index = ures_getSize(dictRes.getAlias());
     fDict = uhash_open(uhash_hashUChars, uhash_compareUChars, nullptr, &status);
     if (U_FAILURE(status)) return;
 
-    ures_resetIterator(fDictRes);
+    ures_resetIterator(dictRes.getAlias());
```

Comment:
optional: You should not need to reset this if you have not iterated over this resource before.

optional*2: You could look at ures_getValueWithFallback() (in https://github.com/unicode-org/icu/blob/main/icu4c/source/common/uresimp.h) for another, more modern, way of reading resource data via ResourceDataValue (sink API, see https://github.com/unicode-org/icu/blob/main/icu4c/source/common/resource.h). `getArray()` then `while (array.getValue(index, value)) { const UChar* str = value.getString(...); ... }`